### PR TITLE
git: set reasonable default features

### PR DESCRIPTION
**Pull request by andir**
###### Description of changes

Git should come with a reasonable set of baseline features. Arguable git-send-email is one of them. Development flows, such as those of Linux, depend on the tool being available.

The difference in closure size between this and the previous version is is negligible. There are no additional build-time dependencies pulled in and the runtime closure difference is about 1%.

```
Estimating rebuild amount by counting changed Hydra jobs (parallel=unset).
    470 x86_64-darwin
    493 x86_64-linux
```

###### Things done

- Built on platform(s)
  - [x] x86_64-linux

### CODE REVIEW

**Review by mweinelt — APPROVED** [review]
Allowing the send-email workflow in the default package seems very reasonable to me. I don't understand why it was disabled in the first place.

**Comment by raphaelr** [comment]
This should target `staging`.

**Comment by bjornfor** [comment]
Mass rebuild -> target staging branch.

**Comment by andir** [comment]
> Mass rebuild -> target staging branch.

That is surprising in multiple ways.

I did run the old maintainers script that supposedly checks rebuild amounts. It didn't look too bad.

Apparently `fetchgit` is using this "fuller" featured git executable instead of a minimal variant that doesn't ship things like git-send-email. There I'd fully agree that it is not needed and shouldn't be required. Maybe worth some follow-up work...

**Comment by andir** [comment]
Actually this change is a no-op for `gitMinimal` which is used as `git` in `fetchgit`. Probably some other packages are picking a fully featured git for no good reason? I'll dive deeper but this shouldn't block this PR.

**Comment by ajs124** [comment]
> Probably some other packages are picking a fully featured git for no good reason

 #206195 might be related

**Comment by Mindavi** [comment]
LGTM, cross-compiled git won't enable send-email but that's fine. Otherwise it wasn't building properly anyway :)

**Review by Mindavi — APPROVED** [review]

**Pull request merged**
